zebra-chain/src/sapling/shielded_data.rs

```diff
@@ -3,10 +3,29 @@ use futures::future::Either;
 use crate::{
     amount::Amount,
     primitives::redjubjub::{Binding, Signature},
-    sapling::{Nullifier, Output, Spend, ValueCommitment},
+    sapling::{tree, Nullifier, Output, Spend, ValueCommitment},
     serialization::serde_helpers,
 };
 
+#[derive(Clone, Debug)]
+pub struct PerSpendAnchor {}
+
+#[derive(Clone, Debug)]
+pub struct SharedAnchor {}
```

Comment:
Since these are zero-sized types, they should also implement `Copy`.

The compiler might implement `Copy` on them automatically, but if not, please make this change:
```suggestion
#[derive(Copy, Clone, Debug)]
pub struct PerSpendAnchor {}

#[derive(Copy, Clone, Debug)]
pub struct SharedAnchor {}
```